mage_ai/data_preparation/models/block/sql/utils/shared.py

```diff
@@ -257,6 +257,17 @@ def table_name_parts(
     return database, schema, table
 
 
+def table_name_parts_from_query(
+    query: str,
+) -> Tuple[str, str, str]:
+    match = re.search(r'[^.+]from[\s](.+)\.(.+)\.(.+)', query, re.IGNORECASE)
```

Comment:
why is there a `from` string in the regex pattern?
also, could you add a unit test for this method?

Reply:
could you also add comment to this method?

Reply:
If the method can also return None, you should set the return type to `Union[Tuple[str, str, str], None]`

Reply:
@wangxiaoyou1993 Added a comment. Also updated the regex to be more flexible (e.g. allow for a WHERE clause). I also added a condition so that we are only parsing the table name parts if the user is NOT using a raw sql query. There might be some unexpected issues with the raw sql query I'm not handling, since this was moreso for users not customizing their own queries and making it more convenient to use SELECT queries specifically (hence the `from` in the regex).

I will add a unit test.